src/main/java/org/peergos/APIService.java

```diff
@@ -42,16 +43,17 @@ public List<HashedBlock> getBlocks(List<Want> wants, Set<PeerId> peers, boolean
                         remoteBlocks.get(remote, peers, addToLocal).stream())
                 .collect(Collectors.toList());
     }
-    
-    public Cid putBlock(byte[] block, String format) {
-        Cid.Codec codec = null;
-        if (format.equals("raw")) {
-            codec = Cid.Codec.Raw;
-        } else if (format.equals("cbor")) {
-            codec = Cid.Codec.DagCbor;
+
+    public boolean accepts(Cid.Codec codec) {
+        if (store instanceof TypeLimitedBlockstore) {
```

Comment:
This class shouldn't depend on a specific blockstore impl. Just parse the codec from the strings.